osu.Game.Rulesets.Osu/Mods/OsuModAlternate.cs

```diff
@@ -1,31 +1,63 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
+using System.Linq;
+using osu.Framework.Bindables;
 using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
 using osu.Game.Rulesets.Mods;
 using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Scoring;
 using osu.Game.Rulesets.UI;
+using osu.Game.Screens.Play;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModAlternate : ModAlternate<OsuHitObject, OsuAction>
+    public class OsuModAlternate : Mod, IApplicableToDrawableRuleset<OsuHitObject>, IApplicableToPlayer, IUpdatableByPlayfield
     {
+        public override string Name => @"Alternate";
+        public override string Acronym => @"AL";
+        public override string Description => @"Don't use the same key twice in a row!";
+        public override double ScoreMultiplier => 1.0;
+        public override Type[] IncompatibleMods => new[] { typeof(ModAutoplay) };
+        public override ModType Type => ModType.DifficultyIncrease;
+        public override IconUsage? Icon => FontAwesome.Solid.Keyboard;
+
+        /// <summary>
+        /// Whether incoming input must be checked by <see cref="InputInterceptor"/> before it is passed to gameplay.
+        /// </summary>
+        public bool CanIntercept => !isBreakTime.Value && introEnded;
+
+        private bool introEnded;
+        private double earliestStartTime;
+        private IBindable<bool> isBreakTime;
         private const double flash_duration = 1000;
         private OsuAction? lastActionPressed;
         private DrawableRuleset<OsuHitObject> ruleset;
 
-        public override void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
+        public void ApplyToDrawableRuleset(DrawableRuleset<OsuHitObject> drawableRuleset)
         {
             ruleset = drawableRuleset;
-            base.ApplyToDrawableRuleset(drawableRuleset);
+            drawableRuleset.KeyBindingInputManager.Add(new InputInterceptor(this));
+
+            var firstHitObject = ruleset.Objects.FirstOrDefault();
+            earliestStartTime = (firstHitObject?.StartTime ?? 0) - (firstHitObject?.HitWindows.WindowFor(HitResult.Meh) ?? 0);
```

Comment:
probably fine for now, but i'm just going to note that if this gets generalised somehow then it should probably be using something like this:

https://github.com/ppy/osu/blob/4132f67629467f5a5f2b2b480b391525bfe7d95e/osu.Game/Rulesets/Scoring/HitWindows.cs#L177